src/coreclr/jit/instr.cpp

```diff
@@ -795,11 +813,104 @@ CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
             assert(op->OperIsHWIntrinsic());
 
 #if defined(FEATURE_HW_INTRINSICS)
-            GenTreeHWIntrinsic* hwintrinsic = op->AsHWIntrinsic();
-            NamedIntrinsic      intrinsicId = hwintrinsic->GetHWIntrinsicId();
-
+            GenTreeHWIntrinsic* hwintrinsic  = op->AsHWIntrinsic();
+            NamedIntrinsic      intrinsicId  = hwintrinsic->GetHWIntrinsicId();
+            var_types           simdBaseType = hwintrinsic->GetSimdBaseType();
             switch (intrinsicId)
             {
+                case NI_AVX_BroadcastScalarToVector128:
+                case NI_AVX_BroadcastScalarToVector256:
+                {
+                    // we have the assumption that AVX_BroadcastScalarToVector*
+                    // only take the memory address as the operand.
+                    assert(hwintrinsic->isContained());
+                    assert(hwintrinsic->OperIsMemoryLoad());
+                    assert(hwintrinsic->GetOperandCount() == 1);
+                    assert(varTypeIsFloating(simdBaseType));
+                    GenTree* broadcastScalar = hwintrinsic->Op(1);
+                    assert(broadcastScalar->isContained());
+                    if (broadcastScalar->OperIs(GT_LCL_ADDR))
+                    {
+                        addr = hwintrinsic->Op(1);
+                        break;
+                    }
+                    else
+                    {
+                        assert(broadcastScalar->OperIs(GT_LCL_VAR));
+                        return OperandDesc(simdBaseType, broadcastScalar);
+                    }
+                }
+
+                case NI_SSE3_MoveAndDuplicate:
+                case NI_AVX2_BroadcastScalarToVector128:
+                case NI_AVX2_BroadcastScalarToVector256:
+                case NI_AVX512F_BroadcastScalarToVector512:
+                {
+                    assert(op->isContained());
+                    if (intrinsicId == NI_SSE3_MoveAndDuplicate)
+                    {
+                        assert(simdBaseType == TYP_DOUBLE);
+                    }
+                    // if broadcast node is contained, should mean that we have some forms like
+                    // broadcast -> CreateScalarUnsafe -> scalar.
+                    // if so, directly emit scalar.
+                    switch (simdBaseType)
+                    {
+                        case TYP_INT:
+                        case TYP_UINT:
+                        case TYP_LONG:
+                        case TYP_ULONG:
+                        {
+                            // a special case is when the operand of CreateScalarUnsafe is in integer type,
+                            // CreateScalarUnsafe node will be fold, so we directly match a pattern of
+                            // broadcast -> LCL_VAR(TYP_(U)INT)
+                            assert(hwintrinsic->Op(1)->OperIs(GT_LCL_VAR, GT_CNS_INT, GT_IND));
```

Comment:
Not sure this assert is quite right. In general we can encounter at least `isIndir()`, `LCL_FLD`, `LCL_VAR`, and `CNS_INT`.

I expect it might be better to specially handle `CNS_INT` and then have the others be handled recursively via `return genOperandDesc(hwintrinsic->Op(1));` since they should all follow the standard containment handling logic.

That helps ensure we don't miss any cases at all.